### ISSUE TITLE
localhost vs. 127.0.0.1

### ISSUE THREAD

**Issue by benknoble** [rmculpepper/racket-http123#4]
I noticed an oddity yesterday, though it may be part of an issue lower in the networking stack that `http123`.

With any server running on localhost (adjust `add-response-handlers` as necessary), the following took roughly 8s (but `,time` only reports 2-4ms of actual work)

```racket
(define client (http-client #:add-response-handlers …))
(send client handle (request 'GET (build-url "http://localhost:XXXX/some-route")))
```

Conversely, changing `localhost` to `127.0.0.1` finished instantly, as expected.

---

Machine details:

- Racket 8.2 CS
```
OS: macOS Catalina 10.15.7 19H1519 x86_64 
Host: MacBookPro16,1 
Kernel: 19.6.0 
CPU: Intel i9-9880H (16) @ 2.30GHz 
GPU: Intel UHD Graphics 630, AMD Radeon Pro 5500M 
```

The `/etc/hosts` includes
```
127.0.0.1	localhost
```

and `NO_PROXY` includes both `localhost` and `127.0.0.1`.

**Comment by rmculpepper** [comment]
My best guess is that it's a lower-level issue, probably related to DNS, but I don't know the details or the right way to fix it. (Other than changing your URLs to use the IP address instead of localhost.) Here's one discussion that seems related: https://www.mail-archive.com/racket-users@googlegroups.com/msg45159.html.

Can you double-check by comparing the following two times (with your server running):

    (time (tcp-connect "localhost" 80))
    (time (tcp-connect "127.0.0.1" 80))

**Comment by benknoble** [comment]
Yes, I did just change the program to use the IP address, since it's a personal program that only needs to work locally at the moment.

After correcting the ports for my server's oddities:
```
> (time (tcp-connect "localhost" 8443))
cpu time: 1 real time: 26081 gc time: 0
#<input-port:localhost>
#<output-port:localhost>
> (time (tcp-connect "127.0.0.1" 8443))
cpu time: 0 real time: 0 gc time: 0
#<input-port:127.0.0.1>
#<output-port:127.0.0.1>
```
